src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -228,12 +252,44 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
     }
 
-    void Actions::_ViewModelRebindKeysHandler(const Editor::KeyBindingViewModel& senderVM, const Editor::RebindKeysEventArgs& args)
+    void Actions::_ViewModelModifyKeyBindingHandler(const Editor::KeyBindingViewModel& senderVM, const Editor::ModifyKeyBindingEventArgs& args)
     {
+        // If the key chord was changed,
+        // update the settings model and view model appropriately
+        auto attemptRebindKeys = [=]() {
+            if (args.OldKeys().Modifiers() != args.NewKeys().Modifiers() || args.OldKeys().Vkey() != args.NewKeys().Vkey())
+            {
+                // update settings model
+                _State.Settings().ActionMap().RebindKeys(args.OldKeys(), args.NewKeys());
+
+                // update view model
+                auto senderVMImpl{ get_self<KeyBindingViewModel>(senderVM) };
+                senderVMImpl->Keys(args.NewKeys());
+            }
+        };
+
+        // If the action was changed,
+        // update the settings model and view model appropriately
+        auto attemptSetAction = [=]() {

```

Comment:
Yup! I really only separated them out so that it's easier to read. I'm down to combine them, but I'm still worried about readability. Would something like this be clear maybe?
```c++
auto applyChangesToSettingsModel = [](){
   // code for 'attemptRebindKeys'
   // code for 'attemptSetAction'
};
```